pennylane/tape/measure.py

```diff
@@ -78,6 +78,16 @@ def __init__(self, return_type, obs=None, wires=None, eigvals=None):
         # Queue the measurement process
         self.queue()
 
+    def __repr__(self):
+        """Representation of this class."""
+        if self.obs is None:
+            return "{}(None)".format(self.return_type.value)
+
+        if self.obs.return_type is None:
+            return "{}({})".format(self.return_type.value, self.obs)
```

Comment:
If the return type is not none, it was already added in the `__repr__` of the observable. 

I am not sure if this has to do with the two cores? Shouldn't the return type be only specified in the process? I tried not to change too much here...

Reply:
> I am not sure if this has to do with the two cores? Shouldn't the return type be only specified in the process? I tried not to change too much here...

Yep, this is to support non-tape mode, where observables are still annotated directly with the return type.

Perhaps we should add a TODO to this method, to remind us to fix it when tape mode is default?